### ISSUE TITLE
Installation stuck after database configuration step.

### ISSUE THREAD

**Issue by xiangpei1028** [olab/Open-Labyrinth#118]
I want to setup a openlabyrinth server on a CentOS 7 system. but each time when the installation process past the database configuration (step 3) the next button will bring the browser into a blank page.

I have tried version 3.4.1 & 3.3.2 & 3.3.1, all the same. I have tried install through firefox & chrome all the same.

there is no errors logged into either apache or php log files neither inside the app's own log files

Please help, this looks like a dead end to me

Thanks

**Comment by xiangpei1028** [comment]
BTW: there 4 steps in total for the installation, but whatever has been entered through the previous 3 steps were not write into any of the configuration files ( i would assume they are sitting in .../Open-Labyrinth/www/application/config), no tables were created on the database side, the app is not even talk to the database at all.

**Comment by topps** [comment]
There are some weirdly coded file paths in OLab3 which often cause problems like this. So an install will work smoothly into default file/folder locations but fail if anything is non-standard about the setup. Olab3 must be the default root in Apache, not a sub-domain etc, which is also a nuisance. There is a workaround to this but it is quirky.

In OLab4, we hope to have resolved all this.


Prof David Topps
OpenLabyrinth Development Consortium Lead

**Comment by xiangpei1028** [comment]
I have now changed the doc root to /var/www/html which is the default document root for apache on Centos 7, i moved the content from /Open-labyrinth/www to /var/www/html and moved /Open-labyrinth/database to /var/www/database, but still getting the same error. and i tried to move /Open-labyrinth/www to /var/www/html/www and /Open-labyrinth/database to /var/www/html/database.

Both approach failed in the same symptom i described before. the installation process just stops right after the database configuration page. click on the next button gives me the blank page, and none of the configuration parameters has been implemented into either the conf file or the database

anything else i could try?

Thanks

**Comment by topps** [comment]
I’ll give it some thought. In the meantime, would you like some free accounts on our demo server so that at least you can get on with writing and testing your virtual scenarios? Let me know via direct email if you would like to do this.

The demo server is fully functional and is not feature limited. The main drawback is that it is a demo/dev server so we do not guarantee uptime. It is, however, pretty stable. We just cannot recommend it for full production, mission critical situations.


Prof David Topps
OpenLabyrinth Development Consortium Lead

**Comment by xiangpei1028** [comment]
Hi David,

Thanks for the offering, we have actually tried your demo server. That's why we would like to get the app and try to setup our own so we could do a little customization to satisfy our environment.

I have one thought, I wonder is there any way we could bypass the installation GUI/Process by directly changing the app configuration files. I think if this method works it will save both of us lots of time.

I have no idea where and what files need to be changed, all i could find is the /application/config directory and bunch of files. I don't know if these are all of the config files or is there any others. And i don't know how to change it since i believe the password part for the GUI admin might involve some encryption mechanism.

On the other hand, i have done some strace to the apache processes and noticed the whole process stopped after the configuration parameters have been written into /var/lib/php directory as php cache session. No idea why it's not writing it to the conf files and database.

Let us know if we could bypass the installation GUI to do a "manual" install of the app

Thanks

**Comment by topps** [comment]
Yes, it is absolutely possible to do that. I have some notes somewhere on it that I can send you. I’ll just have to track them down.

On the customization side, I have a question. For OLab4, we have been looking at i18n internationalization and double byte characters. Do you have a need for that?

dt

**Comment by xiangpei1028** [comment]
Hi David,

We are currently working on Olab3. We would like to have a stable version for the production at the moment. For Olab4, we might do some test after we setup the current environment. For now, let's focus on finding a working "manual installation" procedure at first.

Please let us know when you have found the procedure. I saw some people was trying to do the same and there are some directory & files need to be removed after the installation completes. Please also include these details if possible. It could also help other people who might has the same similar symptoms if we could get a standard "manual installation" procedure 

Many thanks

**Comment by xiangpei1028** [comment]
Hi David,

is there any updates on this ticket yet? We are still waiting 

thanks

Eric
